processor/src/substrate_signer.rs

```diff
@@ -328,19 +338,24 @@ impl<D: Db> SubstrateSigner<D> {
       }
 
       CoordinatorMessage::BatchReattempt { id } => {
-        self.attempt(txn, id.id, id.attempt).await;
+        // TODO: this unwrap should be safe?
+        self.attempt(txn, id.id, id.block.unwrap(), id.attempt).await;
       }
     }
   }
 
-  pub fn batch_signed(&mut self, txn: &mut D::Transaction<'_>, block: BlockHash) {
+  pub fn batch_signed(&mut self, txn: &mut D::Transaction<'_>, batch_id: u32) {
+    // convert into slice
+    let mut id = [0u8; 32];
+    id.copy_from_slice(&batch_id.to_le_bytes());
```

Comment:
ID should be a Transcript of batch and there should be a common function for it.